### ISSUE TITLE
Support recurring events

### ISSUE THREAD

**Issue by osca** [ebarooni/capacitor-calendar#99]
It would be great to offer recurring events with options like:
- recurrence: daily, weekly, monthly, yearly
- recurrenceEndDate: Date when the recurrence 
- recurrenceInterval: number (e.g. every x days)

**Comment by nathanblankson** [comment]
Hey 👋 - just to add here, would it be possible to specify the **recurrence days**? (probably as an addition to supporting simpler recurring events if it's a bit much)

So for example, it would be nice to be able to create a Weekly event that occurs on Monday, Wednesday, Friday.

I think with EventKit it's `EKRecurrenceDayOfWeek` and with Google Calendar there's `BYDAY` in the recurrence rule.

Thanks!

**Comment by ebarooni** [comment]
Hey @nathanblankson ,

Doesnt the initial suggestion already cover the case you described? For example:

- Recurrence frequencey: Weekly
- End date: <SOME_DATE_IN_THE_FUTURE> – Lets say 4 weeks from now
- Interval: Every 1 week

Result: The event is created on a Monday and will be repeated every week on Monday until the end date is reached.

I think the reminders have this functionality implemented already.

**Comment by osca** [comment]
I think not. @nathanblankson means cases like repeat Monday, Tuesday and Thursday events every week.

**Comment by nathanblankson** [comment]
@osca Yeah, that's what I meant - that way, we can build something like this to programmatically set the recurring days

<img width="269" alt="image" src="https://github.com/ebarooni/capacitor-calendar/assets/35101840/9f0e3b21-afa0-4d49-80d8-2e4bda5be154">

**Comment by nathanblankson** [comment]
Hey @osca, hope you're doing well!

Has there been any progress on this piece? Me and my team are quite eager to have this functionality in-app.

Cheers.

**Comment by osca** [comment]
Nothing new heard about this issue. We're using another library for now [@awesome-cordova-plugins/calendar](https://github.com/danielsogl/awesome-cordova-plugins/blob/master/docs/plugins/calendar.md) which lacks other functionality, but we can live with that.

**Comment by vlad31011** [comment]
hi @ebarooni , is it still in your plan to add "Reccuring" support? like its done here https://ionic.io/docs/supported-plugins/calendar#optional-recurrence

**Comment by ebarooni** [comment]
Hi @vlad31011 , this is definitely in the plan but i cant give a time estimate for it. It will be included in the next release.

**Comment by beckerJoel** [comment]
Hi @vlad31011 and @ebarooni, me and a colleague are working on that feature right now. We hope to bring a merge request soon to this repository!

**Comment by AhegaHOE** [comment]
> Hi @vlad31011 and @ebarooni, me and a colleague are working on that feature right now. We hope to bring a merge request soon to this repository!

FYI: I merged the accounts so I'm working on it.

**Comment by ebarooni** [comment]
@AhegaHOE Thanks im looking forward to it. If applicable, could you please distribute your contribution into separate PRs? Such that each PR has a small scope.

**Comment by AhegaHOE** [comment]
@ebarooni Sure, I will try my best to create small PRs for each feature. We are adding quite a bit of stuff! It would probably be better to do it in small steps at a time.
